book/src/chap02.asciidoc

```diff
@@ -1,67 +1,65 @@
+Uma _declaração de atribuição_ cria uma nova variável e da à ela um valor:
 
-An _assignment statement_ creates a new variable and gives it a value:
 (((assignment statement)))((("statement", "assignment", see="assignment statement")))(((pass:[=], see="assignment statement")))(((pi)))((("π", see="pi")))
 
 [source,@julia-repl-test chap02]
 ----
-julia> message = "And now for something completely different"
-"And now for something completely different"
+julia> mensagem = "E agora para algo completamente diferente"
+"E agora para algo completamente diferente"
 julia> n = 17
 17
 julia> π_val = 3.141592653589793
 3.141592653589793
 ----
+Este exemplo faz três atribuições. A primeira atribui uma string à uma nova variável chamada +mensagem+; a segunda da o inteiro +17+ a +n+; a terceira atribui (aproximadamente) o valor de latexmath:[\(\pi\)] a +π_val+ (*+\pi TAB+*).
 
-This example makes three assignments. The first assigns a string to a new variable named +message+; the second gives the integer +17+ to +n+; the third assigns the (approximate) value of latexmath:[\(\pi\)] to +π_val+ (*+\pi TAB+*).
+Um jeito comum de representar variáveis no papel é escrevendo o nome com uma flecha apontando para o seu valor. Este tipo de figura é denominado de _diagrama de estado_ pois mostra em que estado cada variável encontra-se (imagine isso como sendo o estado de espirito da variável) <<fig02-1>> mostra o resultado do exemplo anterior.
 
-A common way to represent variables on paper is to write the name with an arrow pointing to its value. This kind of figure is called a _state diagram_ because it shows what state each of the variables is in (think of it as the variable’s state of mind). <<fig02-1>> shows the result of the previous example.
 (((state diagram)))((("diagram", "state", see="state diagram")))
 
 [[fig02-1]]
-.State diagram
+.Diagrama de estado
 image::images/fig21.svg[]
 
 
-=== Variable Names
+=== Nomes de variáveis
 (((variable)))
+Programadores geralmente escolhem nomes para suas variáveis que são significativos—eles documentam para o que a variável é usada.
 
-Programmers generally choose names for their variables that are meaningful—they document what the variable is used for.
+Nomes de variáveis podem ser o quão longas você desejar. Eles podem conter quase todos os caracteres Unicode (veja <<caracteres>>), mas não podem começar com um número. É permitido usar letras maiúsculas, mas é convencional utilizar somente letras minúsculas para nomes de variáveis.
 
-Variable names can be as long as you like. They can contain almost all Unicode characters (see <<characters>>), but they can’t begin with a number. It is legal to use uppercase letters, but it is conventional to use only lower case for variable names.
-
-Unicode characters can be entered via tab completion of LaTeX-like abbreviations in the Julia REPL.
+Caracteres Unicode podem ser inseridos através do autopreenchimento de abreviações LaTeX no REPL do Julia.
```

Comment:
Aqui vale a pena mudar um pouco o texto para deixar claro que é um completamento feito pelo tab.
Aproveitando, fiquei na dúvida entre autopreencher e autocompletar. Algum motivo pra escolher o primeiro?

Reply:
Agora que li com mais atenção vi que a tradução pode estar errada. "completion of *LaTeX-like* abbreviations...", não estaria fazendo referência a abreviações *do tipo* LaTeX?
Acho que os dois funcionam, mas autopreencher parece "mais correto" na minha opinião. No entanto o nome "oficial" do recurso é autocomplemento, então acredito que este é na verdade o "mais correto".